internal/log/log.go

```diff
@@ -0,0 +1,42 @@
+package log
+
+import (
+	"github.com/gemalto/flume"
+)
+
+var (
+	log flume.Logger
+)
+
+func init() {
+	log = flume.New("upctl")
+}
+
+// SetDebugMode sets debug mode on to get more verbose logs
+func SetDebugMode(v bool) error {
+	if err := flume.Configure(flume.Config{
+		DefaultLevel: flume.DebugLevel,
+		Encoding:     "term-color",
+	}); err != nil {
+		return err
+	}
+	if !v {
+		if err := flume.Configure(flume.Config{
+			DefaultLevel: flume.InfoLevel,
+			Encoding:     "term-color",
+		}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Info Writes info level logs
+func Info(nameSpace string, msg string, args ...interface{}) {
+	log.With("namespace: ", nameSpace).Info(msg, args)
```

Comment:
Is this a flume 'key/value-logging argument'? Eg. does this end up in, say, JSON formatted logs as `{ "namespace: ": "[nameSpace]" }`? I guess the encoding should depend on the output selected as well, or? Not sure, tbh, and that might open up a whole new can of worms.. :o

Reply:
AFAIK with flume you should treat the `args ...interface{}` without any formatting. I'd remove spaces and `:` chars and let flume format the kv-pairs. You can of course implement custom encoder for human readable output.

Reply:
Creating the flume in pckg level. We can input the pckg name in these so the namespace IMO becomes obsolete. We can elaborate more on the location in msg field. I agree with aakso about the formatting of the args. Looks good the way Flume does it.